crates/sui-framework/packages/sui-framework/tests/crypto/bls12381_tests.move

```diff
@@ -382,6 +382,38 @@ module sui::bls12381_tests {
         let _ = bls12381::hash_to_g1(&vector[]);
     }
 
+    #[test]
+    fun test_to_from_uncompressed() {
+        let a = bls12381::g1_generator();
+        let a_uncompressed = bls12381::g1_to_g1_uncompressed(&a);
+        assert!(a_uncompressed.bytes() == x"17f1d3a73197d7942695638c4fa9ac0fc3688c4f9774b905a14e3a3f171bac586c55e83ff97a1aeffb3af00adb22c6bb08b3f481e3aaa0f1a09e30ed741d8ae4fcf5e095d5d00af600db18cb2c04b3edd03cc744a2888ae40caa232946c5e7e1");
+
+        let reconstructed = bls12381::g1_uncompressed_to_g1(&a_uncompressed);
+        assert!(group_ops::equal(&a, &reconstructed));
+    }
+
+    #[test]
+    fun test_uncompressed_sum() {
+        let sum = bls12381::g1_uncompressed_sum(&vector[]);
+        assert!(group_ops::equal(&bls12381::g1_to_g1_uncompressed(&bls12381::g1_identity()), &sum));
+
+        let a = bls12381::g1_generator();
+        let sum = bls12381::g1_uncompressed_sum(&vector[bls12381::g1_to_g1_uncompressed(&a)]);
+        assert!(group_ops::equal(&bls12381::g1_to_g1_uncompressed(&a), &sum));
+
+        let b = bls12381::g1_mul(&bls12381::scalar_from_u64(2), &bls12381::g1_generator());
```

Comment:
let's instead use random scalars and a random length of terms below

Reply:
you can use random parameters in move tests, e.g.,
`#[random_test]
fun div_by(x: u64) {
   x / (x % 10)
}`